File: WebClientIS/WebClientIS/Models/RelSin.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;

namespace APIIS.Models
{
    public class RelSin
    {
        public int IdP { get; set; }

        public int IdS { get; set; }

        public DateTime DataSintoma { get; set; }

        public RelSin()
        {
            IdP = 0;
            IdS = 0;
            DataSintoma = new DateTime();

        }


        public List<RelSin> GetRelSins(int idp)
        {
            List<RelSin> lista = new List<RelSin>();

            HttpClient client = new HttpClient();

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/RelSin/"+idp.ToString());

            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

            HttpResponseMessage response = client.GetAsync("").Result;

            if (response.IsSuccessStatusCode)
            {
                var products = response.Content.ReadAsStringAsync().Result;

                var array = JArray.Parse(products);

                foreach (var obj in array)
                {
                    try
                    {
                        lista.Add(obj.ToObject<RelSin>());
                    }
                    catch (Exception ex)
                    {
                        return new List<RelSin>();
                    }
                }

            }

            return lista;
        }

        public bool CreateRelSin()
        {
            HttpClient client = new HttpClient();

            var values = new Dictionary<string, object>
            {
                { "IdP", IdP.ToString() },
                {"IdS",IdS.ToString() },
                {"DataSintoma",DataSintoma }


            };

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/RelSin");

            var jsonString = JsonConvert.SerializeObject(values);
            var content = new StringContent(jsonString, Encoding.UTF8, "application/json");

            var resposta = client.PostAsync(client.BaseAddress.ToString(), content).Result;

            if (resposta.IsSuccessStatusCode)
            {
                return true;
            }
            else
            {
                return false;
            }
        }
    }
}

File: MobileClient/sampleproject/lib/Models/RelSin.dart
import 'dart:typed_data';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert' show json;
import 'package:scoped_model/scoped_model.dart';


class RelSin extends Model{

  int idp;
  int ids;
  DateTime datasintoma;

  RelSin({int idpp,int idss,DateTime ds})
  {
    this.idp = idpp;
    this.ids = idss;
    this.datasintoma = ds;
  }

  factory RelSin.fromJson(Map<String,dynamic> json){

    return RelSin(
      idpp: json['idP'] as int,
      idss: json['idS'] as int,
      ds:DateTime.parse(json['dataSintoma'])
      
    );

  }

  Future<List<RelSin>> getrelsin() async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/RelSin"),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<RelSin> listau = new List<RelSin>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(RelSin.fromJson(lista[i]));
  

  return listau;
}

Future<List<RelSin>> getrelsinidp(int idp) async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/RelSin/"+idp.toString()),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<RelSin> listau = new List<RelSin>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(RelSin.fromJson(lista[i]));
  

  return listau;
}

Future<int> createRelSin(List<RelSin> relSin) async {

  var url ='http://6f85a52b4f05.ngrok.io/api/RelSin';

   http.Response response;

   int count = 0;

  for(RelSin i in relSin)
  {
    var element = json.encode(<String, String>{
        'IdP': i.idp.toString(),
        'IdS':i.ids.toString(),
        'DataSintoma':i.datasintoma.toString()    

      });


      response = await http.post(url,
      headers: {"Content-Type": "application/json"},
      body: element
  );

  if(response.statusCode == 200)
  {
    count++;
  }

  }

  if(count == relSin.length)
  {
    return 1;
  }
  else
  {
    return 0;
  }
}

}